zpriddy/SmartThings_PyDash
# CSS refactor, adding grunt.js

# Summary

This pull request offers a cleanup of stylesheet files (as well as how they are organized/handled) and the addition of grunt.js to concatenate and compile SCSS.
# Background

When I first pulled down the project and started to poke around, I had trouble figuring out where to implement style changes because there were some CSS files that weren't being used, and lots of duplication of styles. For example, most of the individual widget styles had been copied out of their individual SCSS files and placed in application.css, rather than allowing the style rules for each widget to remain primarily driven by that widget's individual .scss file. Additionally, the application.scss file seemed to be ignored, in favor of just making direct edits in application.css.
# Goal

To make it easy to add widgets in the future, I wanted to adjust the project so it respected and incorporated each widget's SCSS file. I wanted to reduce duplication of style rules scattered throughout the project, and make it easier to determine where style edits should be made.
# What's changed
1. grunt is now required to compile the CSS for the project. NPM is used to install grunt and dependencies.
2. application.css itself has been removed from the repo. Instead, the project now relies on SCSS files. The grunt tasks (found in Gruntfile.js) search through the project, find all the SCSS files, concatenate them, and then compiles them into a nice big application.css file.
3. The /assets/application.css route in main.py has been changed to reflect the above: that only the application.css file gets loaded when requested by the browser.
4. I went through and individually tweaked styles in a few of the widgets' SCSS files to get them to match more closely the look that the widgets had before I started in on this little CSS-scalping adventure.
# How to test

Pull down this branch and run it according to the updated README. Check that everything still works as before and more or less looks right.
